gym_hil/wrappers/intervention_utils.py

```diff
@@ -14,6 +14,88 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+import os
+import logging
+from pathlib import Path
+
+
+def load_controller_config(controller_name=None, config_path=None):
+    """
+    Load controller configuration from JSON file.
+    
+    Args:
+        controller_name: Name of the controller to load configuration for.
+                         If None, will use 'default' configuration.
+        config_path: Path to the controller configuration JSON file.
+                     If None, will look for controller_config.json in multiple locations.
+    
+    Returns:
+        Dictionary containing controller configuration.
+    """
+    # List of paths to check for the configuration file
+    paths_to_check = []
+    
+    # If a specific path is provided, check it first
+    if config_path is not None:
+        paths_to_check.append(Path(config_path))
+    
+    # Check in the current working directory
+    paths_to_check.append(Path.cwd() / "controller_config.json")
+    
+    # Check in the gym_hil package directory
+    module_dir = Path(__file__).parent.parent
+    paths_to_check.append(module_dir / "controller_config.json")
+    
+    # Try each path in order
+    config = None
+    for path in paths_to_check:
+        try:
+            with open(path, 'r') as f:
+                config = json.load(f)
+                logging.info(f"Loaded controller configuration from {path}")
+                break
+        except (FileNotFoundError, json.JSONDecodeError):
+            continue
+    
+    # If no configuration file was found, use default values
+    if config is None:
+        logging.warning("No controller configuration file found. Using default values.")
+        config = {
+            "default": {
+                "axes": {
+                    "left_x": 0,
+                    "left_y": 1,
+                    "right_x": 2,
+                    "right_y": 3
+                },
+                "buttons": {
+                    "a": 0,
+                    "b": 1,
+                    "x": 2,
+                    "y": 3,
+                    "lb": 4,
+                    "rb": 5,
+                    "lt": 6,
+                    "rt": 7
+                },
+                "axis_inversion": {
+                    "left_x": False,
+                    "left_y": True,
+                    "right_x": False,
+                    "right_y": True
+                }
+            }
+        }
+    
+    # If controller_name is None or not found in config, use default
+    if controller_name is None or controller_name not in config:
+        if controller_name is not None and controller_name not in config:
+            logging.warning(f"Controller '{controller_name}' not found in config. Using default configuration.")
+        return config.get("default", {})
+    
+    return config[controller_name]
```

Comment:
This function is a bit complicated. Shouldn’t it be simpler? If the path is None, it returns the default config; otherwise, it loads the provided JSON.
Maybe I didn’t fully understand how it works. Could you add a high-level comment explaining the function’s behavior?